tck/features/MatchAcceptance2.feature

```diff
@@ -0,0 +1,2023 @@
+      CREATE (a)-[:CONTAINS]->(b),
+             (b)-[:CONTAINS]->(c),
+             (c)-[:CONTAINS]->(d)
+      """
+    When executing query:
+      """
+      MATCH (a {name: 'A'})-[*]->(x)
+      RETURN x
+      """
+    Then the result should be:
+      | x             |
+      | ({name: 'B'}) |
+      | ({name: 'C'}) |
+      | ({name: 'D'}) |
+    And no side effects
+
+  Scenario: Variable length relationship without lower bound
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}), (b {name: 'B'}),
+             (c {name: 'C'})
+      CREATE (a)-[:KNOWS]->(b),
+             (b)-[:KNOWS]->(c)
+      """
+    When executing query:
+      """
+      MATCH p = ({name: 'A'})-[:KNOWS*..2]->()
+      RETURN p
+      """
+    Then the result should be:
+      | p                                                               |
+      | <({name: 'A'})-[:KNOWS]->({name: 'B'})>                         |
+      | <({name: 'A'})-[:KNOWS]->({name: 'B'})-[:KNOWS]->({name: 'C'})> |
+    And no side effects
+
+  Scenario: Variable length relationship without bounds
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}), (b {name: 'B'}),
+             (c {name: 'C'})
+      CREATE (a)-[:KNOWS]->(b),
+             (b)-[:KNOWS]->(c)
+      """
+    When executing query:
+      """
+      MATCH p = ({name: 'A'})-[:KNOWS*..]->()
+      RETURN p
+      """
+    Then the result should be:
+      | p                                                               |
+      | <({name: 'A'})-[:KNOWS]->({name: 'B'})>                         |
+      | <({name: 'A'})-[:KNOWS]->({name: 'B'})-[:KNOWS]->({name: 'C'})> |
+    And no side effects
+
+  Scenario: Returning bound nodes that are not part of the pattern
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}), (b {name: 'B'}),
+             (c {name: 'C'})
+      CREATE (a)-[:KNOWS]->(b)
+      """
+    When executing query:
+      """
+      MATCH (a {name: 'A'}), (c {name: 'C'})
+      MATCH (a)-->(b)
+      RETURN a, b, c
+      """
+    Then the result should be:
+      | a             | b             | c             |
+      | ({name: 'A'}) | ({name: 'B'}) | ({name: 'C'}) |
+    And no side effects
+
+  Scenario: Two bound nodes pointing to one
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a {name: 'A'}), (b {name: 'B'}),
+             (x1 {name: 'x1'}), (x2 {name: 'x2'})
+      CREATE (a)-[:KNOWS]->(x1),
+             (a)-[:KNOWS]->(x2),
+             (b)-[:KNOWS]->(x1),
+             (b)-[:KNOWS]->(x2)
+      """
+    When executing query:
+      """
+      MATCH (a {name: 'A'}), (b {name: 'B'})
+      MATCH (a)-->(x)<-->(b)
+      RETURN x
+      """
+    Then the result should be:
+      | x              |
+      | ({name: 'x1'}) |
+      | ({name: 'x2'}) |
+    And no side effects
+
+  Scenario: Three bound nodes pointing to one
```

Comment:
"Three bound nodes pointing to one" -> "Three bound nodes pointing to a single node"